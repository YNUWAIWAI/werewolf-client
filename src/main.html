@(title: String, stylesheet: String)(content: Html)
<!DOCTYPE html>
<html lang="ja">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <meta http-equiv="X-UA-Compatible" content="ie=edge">
  <link rel="shortcut icon" type="image/x-icon" href='@routes.Assets.at("images/favicon.ico")'>
  <link rel="stylesheet" href='@routes.Assets.at("stylesheets/" + stylesheet)'>
  <title>@title</title>
</head>
<body>
  @content
</body>
</html>
