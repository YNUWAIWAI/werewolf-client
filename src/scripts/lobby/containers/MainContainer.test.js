// @flow
import * as ActionTypes from '../constants/ActionTypes'
import MainContainer from './MainContainer'
import React from 'react'
import {initialState} from '../reducers/main'
import {shallow} from 'enzyme'

test('<MainContainer /> initialState', () => {
  const dispatch = jest.fn()
  const getState = () => ({
    main: initialState
  })
  const subscribe = jest.fn()
  const store = {
    dispatch,
    getState,
    subscribe
  }
  const wrapper = shallow(<MainContainer store={store} transition={jest.fn()} />)

  expect(wrapper.props().menuItems)
    .toEqual([
      {
        id: 'Menu.lobbyForAudience',
        types: [ActionTypes.SHOW_LOBBY_FOR_AUDIENCE]
      },
      {
        id: 'Menu.lobbyForRobotPlayer',
        types: [ActionTypes.SHOW_LOBBY_FOR_ROBOT_PLAYER]
      },
      {
        id: 'Menu.lobbyForHumanPlayer',
        types: [ActionTypes.SHOW_LOBBY_FOR_HUMAN_PLAYER]
      },
      {
        id: 'Menu.history',
        types: [ActionTypes.SHOW_HISTORY]
      },
      {
        id: 'Menu.settings',
        types: [ActionTypes.SHOW_SETTINGS]
      }
    ])
})
