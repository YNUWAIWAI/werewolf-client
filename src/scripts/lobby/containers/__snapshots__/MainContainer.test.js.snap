// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainContainer /> initialState 1`] = `"<ul class=\\"menu\\"><li class=\\"menu--item  \\">Lobby for audiences</li><li class=\\"menu--item  \\">Lobby for robot players</li><li class=\\"menu--item  \\">Lobby for human players</li><li class=\\"menu--item  \\">History</li><li class=\\"menu--item  \\">Settings</li><li class=\\"menu--item  \\">How to play</li><li class=\\"menu--item  \\">Credits</li></ul>"`;
